### ISSUE TITLE
Support for Anomaly 

### ISSUE THREAD

**Issue by ghost** [Swartz27/xray-16#1373]
**Is your feature request related to a problem? Please describe.**
I would like to create a new branch and adapt this engine for Stalker Anomaly, since its own fork is only contains game-logic changes but the performance is awful.

**Describe the solution you'd like**
Create new branch `/anomaly` and apply/refactor changes from here: https://bitbucket.org/anomalymod/xray-monolith/commits/ to make this fork compatible with **Anomaly** mod.

**Additional context**
I believe this will benefit both the playerbase and the team of this fork, due to the increased popularity.

**Comment by Xottab-DUTY** [comment]
Do you need any help with that?

@ChugunovRoman was working on merging Call of Chernobyl to the latest OpenXRay. Anomaly is based on CoC, so Roman's work can help really much!

**Comment by ghost** [comment]
@Xottab-DUTY Yes, I could use some help. 
I plan to investigate conflicting parts of the code between these two to see where code cannot be simply copy-pasted from anomaly sources. There will be needed merging and refactoring.
I think at first it would be correct to transfer only the mod logic, without touching parts related to the core engine's functionality.

**Comment by LVutner** [comment]
I suggest to compare Anomaly code against: https://bitbucket.org/SeargeDP/coc-1.4-xray-64/src
Then, you can see what was added/removed/changed. Should be easier to port Anomaly functionality onto OpenXRay engine.

**Comment by LVutner** [comment]
Most of the render code (excluding features required/accessed by LUA) can be ignored.

**Comment by Xottab-DUTY** [comment]
@ChugunovRoman, could you give a tip about branches in your fork?
https://github.com/ChugunovRoman/xray-16

I see there are `coc` and `future_changes_for_coc` branches. Which should be used and what changes they contain?

**Comment by ChugunovRoman** [comment]
> I see there are `coc` and `future_changes_for_coc` branches. Which should be used and what changes they contain?

@Xottab-DUTY The `future_changes_for_coc` branch include all latest changes which I moved from the CoC Engine

**Comment by yohjimane** [comment]
This may have some overlap with the work I'm doing for #1398, as I have to port some anomaly functionality to support the shaders

**Comment by mbehm** [comment]
> I suggest to compare Anomaly code against: https://bitbucket.org/SeargeDP/coc-1.4-xray-64/src Then, you can see what was added/removed/changed. Should be easier to port Anomaly functionality onto OpenXRay engine.

I don't know if this has gotten anywhere since the last comments here but I did take a quick look on doing the comparison. A quick initial diff was around million lines, after heavy filtering of all whitespace etc. changes we're left with about 55k lines of diffs that should contain all the actually meaningful changes, here's a repo for the generation and filtered diffs https://github.com/mbehm/xray-monolith-diffs

**Comment by mbehm** [comment]
Small status update

I've created a branch with all the filtered diffs applied to a copy of the base coc src, you can find [the changes here](https://github.com/mbehm/xray-monolith-diffs/compare/master...applied).

Next I'll start to go through the individual change hunks and split them to feature/lua-scripting etc. branches. Starting with misc cleanup etc. to get them out of cluttering the rest of the logic changes.

There's about 2.3k hunks so not too bad but I'm not sure how much time I'll have to work on it so might take few weeks to sort through them.

**Comment by Xottab-DUTY** [comment]
@mbehm, one thing to make sure: did you compared Anomaly against [SeargeDP's CoC x64](https://bitbucket.org/SeargeDP/coc-1.4-xray-64) repository?

**Comment by mbehm** [comment]
> @mbehm, one thing to make sure: did you compared Anomaly against [SeargeDP's CoC x64](https://bitbucket.org/SeargeDP/coc-1.4-xray-64) repository?

Yes the diffs are based on that and the current demonized xray-monolith repository.

**Comment by Xottab-DUTY** [comment]
I created a new task for Call of Chernobyl support (#1529). We should solve it first, before attempting to launch Anomaly.

**Comment by ghost** [comment]
Related https://github.com/OpenXRay/xray-16/issues/1567
